apps/api/src/routers/admin.py

```diff
@@ -125,12 +125,41 @@ async def release_decisions() -> None:
         group = [record for record in records if record["decision"] == decision]
         if not group:
             continue
-
         await asyncio.gather(
             *(_process_batch(batch, decision) for batch in batched(group, 100))
         )
 
 
+@router.post(
+    "/confirm-attendance", dependencies=[Depends(require_role([Role.DIRECTOR]))]
+)
+async def confirm_attendance() -> None:
+    """Update applicant status to void or attending based on their current status."""
+    records = await mongodb_handler.retrieve(
+        Collection.USERS,
+        {"role": Role.APPLICANT},
+        ["_id", "status"],
+    )
+
+    for record in records:
+        _set_confirmations(record)
+
+    for cur_status in (Status.ATTENDING, Status.VOID):
+        group = [record for record in records if record["status"] == cur_status]
+        await asyncio.gather(
+            *(_process_status(batch, cur_status) for batch in batched(group, 100))
+        )
```

Comment:
Issue: While this code properly changes the status of every applicant that has `CONFIRMED`, `WAIVER_SIGNED`, or `ACCEPTED`, it runs update queries for applicants that already have a status of `ATTENDING` or `VOID`, which leads to unnecessary queries sent to the database. To cut down on the number of queries, could you iterate only over the statuses that need to be changed and perform updates on those?

Reply:
I have updated the logic, however, now it becomes pretty much untestable as it's now all inside the router function, but I'm not sure if we truly require it to be tested

I have commented the previous tests and helper functions out for now